### PR TITLE
Feature/uppsf 5001 opa integration

### PR DESCRIPTION
**Pull request by agrancharova**
# Description

## What

Implement authorization based on publicationId and access headers
## Why

[UPPSF-5001](https://financialtimes.atlassian.net/browse/UPPSF-5001)
[UPPSF-4825](https://financialtimes.atlassian.net/browse/UPPSF-4825)
## Anything, in particular, you'd like to highlight to reviewers

Mention here sections of code which you would like reviewers to pay extra attention to .E.g

_Would appreciate a second pair of eyes on the test_  
_I am not quite sure how this bit works_  
_Is there a better library for doing x_  

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [x] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

## DoD - Ensure all relevant tasks are completed before marking this PR as "Ready for review"

- [ ] Test coverage is not significantly decreased
- [x] All PR checks have passed
- [x] Changes are deployed on dev before asking for review
- [x] Documentations remains up-to-date
  - [x] OpenAPI definition file is updated
  - [x] README file is updated
  - [ ] Documentation is updated in upp-docs and upp-public-docs
  - [ ] Architecture diagrams are updated

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

[UPPSF-5001]: https://financialtimes.atlassian.net/browse/UPPSF-5001?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UPPSF-4825]: https://financialtimes.atlassian.net/browse/UPPSF-4825?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/service_test.go [hunk 1]
```diff
@@ -439,19 +439,17 @@ func TestOpaPolicyValidation(t *testing.T) {
 		expectedErrorMessage string
 	}{
 		{
-			name:               "(1)Test opa policy validation authorized with basic authentication no publication id",
+			name:               "(1)Test opa policy validation authorized with basic authentication(no access from header) no publication id",
```
Comment (angelraynovft) [inline content/service_test.go:442]: question | non-blocking: Is there a specific reason to add numbers in the test names?

**Review by angelraynovft — APPROVED** [review]

File: content/service_test.go (hunk 1 above)
Comment (agrancharova) [reply content/service_test.go:442]: my  OCD to count things, and also very easy for me to navigate, when any is failing

**Review by asparuhft — APPROVED** [review]

File: docker-compose-tests.yml [hunk 2]
```diff
@@ -10,14 +10,60 @@ services:
...
+    depends_on:
+      - neo4j
+      - opa
+
+  opa:
+    image: coco/cm-open-policy-agent:latest
+    environment:
+      GITHUB_USERNAME: "${GITHUB_USERNAME}"
+      GITHUB_TOKEN: "${GITHUB_TOKEN}"
+      BRANCH_NAME: "feature/UPPSF-4977-public-content-by-concept-api-policies"
```
[the remark this answers is not in the document]
Comment (agrancharova) [reply docker-compose-tests.yml:65]: Yes, I will change it, now that it is merged

**Review by asparuhft — APPROVED** [review]

**Review by angelraynovft — APPROVED** [review]

**Review by antoanFT — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -3,45 +3,45 @@ module github.com/Financial-Times/public-content-by-concept-api/v2
...
 	github.com/Financial-Times/api-endpoint v1.0.0
 	github.com/Financial-Times/base-ft-rw-app-go/v2 v2.0.0-20211207113529-d0faaa432d5d
 	github.com/Financial-Times/cm-neo4j-driver v1.1.1
-	github.com/Financial-Times/concepts-rw-neo4j v1.36.0
-	github.com/Financial-Times/content-rw-neo4j/v3 v3.5.2
+	github.com/Financial-Times/concepts-rw-neo4j v1.37.1
 	github.com/Financial-Times/go-fthealth v0.0.0-20200609161010-4c53fbef65fa
 	github.com/Financial-Times/go-logger/v2 v2.0.1
 	github.com/Financial-Times/http-handlers-go/v2 v2.3.0
+	github.com/Financial-Times/opa-client-go v1.1.0
```
Comment (tsvetko24) [inline go.mod:14]: Great! :)

**Review by tsvetko24 — APPROVED** [review]

**Review by angelraynovft — APPROVED** [review]

**Review by tsvetko24 — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ Options:
 * `curl http://localhost:8080/content/http://api.ft.com/things/dbb0bdae-1f0c-11e4-b0cb-b2227cce2b54/implicitly `
 
 ## API definition
-Based on the following [google doc](https://docs.google.com/a/ft.com/document/d/1YjqNYEXkc0Ip-6bGttwnPcAh2XKG6tgzmojTdq8gM2s)
+Full API definition and description of supported endpoints can be found in the [Open API specification](./api/api.yml).
```
Comment (tsvetko24) [inline README.md:54]: Thought: Should you add CODEOWNERS file with Metadata-Team to add reviewers for PRs out of the box...
/.github/CODEOWNERS
```
# This repo is owned by the universal-publishing team, which is  administered by the FT Content Programme Team.
* @Financial-Times/universal-publishing

# This repo is supported by:
* @Financial-Times/metadata-team
```

Comment (agrancharova) [reply README.md:54]: probably, I will handle this one separately if makes sense - it is not really related to the task

**Pull request merged**